src/sync/sync_permission.cpp

```diff
@@ -61,6 +61,38 @@ Permission::AccessLevel extract_access_level(Object& permission, CppContext& con
 
 }
 
+// MARK: - Permission
+
+std::string Permission::description_for_access_level(AccessLevel level)
+{
+    switch (level) {
+        case AccessLevel::None: return "none";
+        case AccessLevel::Read: return "read";
+        case AccessLevel::Write: return "write";
+        case AccessLevel::Admin: return "admin";
+    }
+    REALM_UNREACHABLE();
+}
+
+bool Permission::paths_are_equivalent(std::string path_1, std::string path_2,
```

Comment:
Is it legal for `~` to appear anywhere other than at the beginning with `/~/`? For example, can I access a Realm at URL `/rootRealmWithUserID_~`? Where the trailing `~` would be substituted with the user ID?

This function replaces _any_ `~` in the path, not just leading `/~/`.